lib/taurus/qt/qtgui/base/taurusbase.py

```diff
@@ -1243,6 +1243,48 @@ def __init__(self, name, parent=None, designMode=False):
         self.call__init__(TaurusBaseComponent, name,
                           parent=parent, designMode=designMode)
         self._setText = self._findSetTextMethod()
+        formatter = getattr(taurus.tauruscustomsettings,
+                            'DEFAULT_FORMATTER', None)
+        if formatter is not None:
+            self._setFormatter(formatter)
+
+    def _getFormatter(self, formatter):
+        '''' Method to get custom formatter
+        :return: formatter: python fromat string or formatter callable
+        (in string version)
+        '''
+        try:
+            moduleName, formatterName = formatter.rsplit('.', 1)
+            __import__(moduleName)
+            module = sys.modules[moduleName]
+            formatter = getattr(module, formatterName)
+        except:
+            pass
+        finally:
+            return formatter
+
+    def showFormatterDlg(self):
+        '''
+        showFormatterDlg show a dialog to get the formatter from the user.
+        :return: formatter: python fromat string or formatter callable
+        (in string version) or None
+        '''
+        formatter, ok = Qt.QInputDialog.getText(self, "Set formatter",
```

Comment:
The dialog should show the current formatter.
And ideally the e dialog could be implemented with an editable combobox  (a Qt.QInputDialog.getItem) instead of a lineedit and pre-populate it with supported formatters.

Reply:
I  believe that the first requirement can be easily done.
But the use of combobox (for the moment) is very complicated since the formaters are not registered (as plugin ) and we allow to use custom external modules.

Reply:
I agree. let's do the first part and leave a TODO for the second